src/lib/components/ClassSchedule.svelte

```diff
@@ -186,65 +167,173 @@
+   * @param link The link to the class session
+   */
+  const recordClass = async (classId: string) => {
+    const {meetingLink} = values;
+    const classDoc = await getDoc(doc(db, classesCollection, classId))
+    if (classDoc.exists()) {
+      const confirmHoldClass = confirm(
+        `Please confirm you are holding class now. Confirming will redirect you to ${meetingLink}`,
       )
-      .join(', ')
-
-    navigator.clipboard
-      .writeText(emailList)
-      .then(() => {
-        alert.trigger('success', 'Emails copied to clipboard!')
-      })
-      .catch((err) => {
-        alert.trigger('error', 'Failed to copy emails to clipboard!')
-      })
+      if (confirmHoldClass) {
+        const data = classDoc.data()
+        let datesHeld: Date[] = []
+        if (!classTodayHeld(data.datesHeld))
+          datesHeld = [...data.datesHeld, new Date()]
+        let classToday = false
+        if (
+          nextClassIndex !== -1 &&
+          nextClassIndex < meetingTimes.length &&
+          new Date().toDateString() ===
+            meetingTimes[nextClassIndex].toDateString()
+        ) {
+          classToday = true
+          values.classStatuses[nextClassIndex] = values.feedbackCompleted[nextClassIndex]
+            ? ClassStatus.EverythingComplete
+            : ClassStatus.FeedbackIncomplete
+        }
+        if (!classToday) {
+          alert.trigger(
+            'error',
+            'No class session found today! Please update your class schedule if you are planning to hold class today.',
+          )
+          return;
+        } else {
+          updateDoc(doc(db, classesCollection, classId), {
+            datesHeld: datesHeld,
+            classesStatus: values.classStatuses,
+          })
+        }
+        window.open(meetingLink)
+      }
+    } else {
+      alert.trigger('error', 'Class Not Found!')
+    }
   }
+
+  function copyEmails() {
+  const emailList = studentList
+    .map(
+      (student) =>
+        `${student.email}${
+          student.secondaryEmail ? `, ${student.secondaryEmail}` : ''
+        }`,
+    )
+    .join(', ')
+
+  navigator.clipboard
+    .writeText(emailList)
+    .then(() => {
+      alert.trigger('success', 'Emails copied to clipboard!')
+    })
+    .catch((err) => {
+      alert.trigger('error', 'Failed to copy emails to clipboard!')
+    })
+}
+
+onMount(() => {
+    return user.subscribe((user) => {
+      if (user) {
+        classId = user.object.uid
+        const classDocRef: DocumentReference = doc(
+          db,
+          classesCollection,
+          classId,
+        )
+        getDoc(classDocRef).then((classDoc) => {
+          if (classDoc.exists()) {
+            const data = classDoc.data()
+            const studentUids = data?.students
+            if (studentUids) {
+              getStudentList(studentUids)
+            }
+            if (data && data.meetingTimes) {
+              meetingTimes = data.meetingTimes.map(
```

Comment:
can we make this a function, there are many places in our codebase where we convert firebase time format to our time format